app/controllers/users/profile_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Users::ProfileController < ApplicationController
+
+# /users/profile/edit
+def edit_image
+  avatar = params[:user][:avatar]
+
+  if avatar != nil
+    begin
+      current_user.avatar = avatar
+      current_user.avatar.recreate_versions!
+      current_user.save!
+      flash[:notice] = "You have successfully set your profile image!"
+    rescue 
+      flash[:notice] = "Please submit an image file"
+    end
+  else
+    flash[:notice] = "There was an issue setting your profile image"
```

Comment:
Shouldn't this and line 14 be switched because if avatar is nil that means an image file was not submitted right?

Reply:
Line 14 runs when a file other than an image file (.txt for example) is submitted.  I think the first error is appropriate, maybe 17's message should be different if you have any ideas.

Reply:
ah ok i just got confused because i thought line 14 meant no file was submitted at all. In that case that's fine. I'm not sure about all the different cases here but seems like line 17 is the case that the parameter does not exist, so if that's the case you could maybe change the error message there to like "you did not submit a profile image!" Unless i'm wrong in which case just leave it.